CryptoScan.Web/client-app/src/App.tsx

```diff
@@ -1,47 +1,109 @@
-import React, {useState} from 'react';
-import logo from './logo.svg';
+import {useState} from 'react';
+import { Symbol, Subscription, TimeRange, Trend } from './Models';
+import {post, get, del, patch} from './Http';
 import './App.css';
+import SubscriptionPanel from "./SubscriptionPanel";
+import { 
+  Box,
+  TableRow,
+  TableCell,
+  TextField,
+  Button,
+  Divider
+ } from '@material-ui/core';
 
-function App() {
-  
-  let [successText, setSuccessText] = useState("");
+ export default function App() {
+
+  const quoteAsset : string = "EUR";
   
-  let onButtonClick = async () => {
-    let response = await fetch("test", {
-      method: "POST",
-      headers: {
-        'content-type': 'application/json;charset=UTF-8'
-      }, 
-      body: ""});
-    let text = await response.text();
-    setSuccessText(text);
+  let [symbols, setSymbols] = useState<Symbol[]>();
+  let [subscriptions, setSubscriptions] = useState<Subscription[]>();
+  let [isEmailSet, setIsEmailSet] = useState(false);
+  let [email, setEmail] = useState("")
+
+  let confirmEmail = () => {
+    setIsEmailSet(true);
+    getAvilableSymbols()
+    getSubscriptions()
+  }
+
+  let getAvilableSymbols = async () => {
+    let response = await get("info/exchange/symbols/" + quoteAsset);
+    let symbols = (await response.json()) as Symbol[];
+    setSymbols(symbols);
+  }
+
+  let getSubscriptions = async () => {
+    let response = await get("info/subscriptions");
+    let subscriptions = (await response.json()) as Subscription[];
+    setSubscriptions(subscriptions);
+  }
+
+  let subscribe = async (newSubscription: Subscription) => {
+    await post("subscriptions/subscribe", JSON.stringify(newSubscription));
+    let newSubscriptions = [...subscriptions as Subscription[], newSubscription];
```

Comment:
is the "as Subscription[]" needed? Seems weird

Reply:
Unfortunately, because otherwise it treats it as Subscription[] | undefined